ap_sta_config.sh

```diff
@@ -189,178 +188,115 @@ if test true != "${STA_ONLY}" && test true == "${AP_ONLY}"; then
 
     # keep order of dependencies installation
     if [[ $(dpkg -l | grep -c cron) == 0 ]]; then
-        apt-get -y update
-        apt-get -y install cron
-    fi
-
-    if [[ $(dpkg -l | grep -c dhcpcd) == 0 ]]; then
-        apt-get -y update
-        apt-get -y install dhcpcd
-    fi
-
-    if [[ $(dpkg -l | grep -c hostapd) == 0 ]]; then
-        apt-get -y update
-        apt-get -y install hostapd
-    fi
-
-    if [[ $(dpkg -l | grep -c dnsmasq) == 0 ]]; then
-        apt-get -y update
-        apt-get -y install dnsmasq
+        apt -y update
+        apt -y install cron
+        apt -y install dhcpcd
+        apt -y install hostapd
+        apt -y install dnsmasq
     fi
 fi
 
 if test true != "${STA_ONLY}"; then
-    # Populate `/etc/udev/rules.d/70-persistent-net.rules`
-    _logger "Populate /etc/udev/rules.d/70-persistent-net.rules"
-    bash -c 'cat > /etc/udev/rules.d/70-persistent-net.rules' <<EOF
-SUBSYSTEM=="ieee80211", ACTION=="add|change", ATTR{macaddress}=="${MAC_ADDRESS}", KERNEL=="phy0", \
-RUN+="/sbin/iw phy phy0 interface add ap0 type __ap", \
-RUN+="/bin/ip link set ap0 address ${MAC_ADDRESS}
-
+    # Exclude ap0 from `/etc/dhcpcd.conf`
+    sudo bash -c 'cat >> /etc/dhcpcd.conf' << EOF
+# this defines static addressing to ap@wlan0 and disables wpa_supplicant for this interface
+interface ap@wlan0
+    static ip_address=${AP_IP}/24
+    ipv4only
+    nohook wpa_supplicant
 EOF
-fi
 
-if test true != "${STA_ONLY}"; then
     # Populate `/etc/dnsmasq.conf`
     _logger "Populate /etc/dnsmasq.conf"
-    bash -c 'cat > /etc/dnsmasq.conf' <<EOF
-interface=lo,ap0
+    bash -c 'cat > /etc/dnsmasq.conf' << EOF
+interface=lo,ap0@wlan0
 no-dhcp-interface=lo,wlan0
 bind-interfaces
-server=8.8.8.8
+server=1.1.1.1
 domain-needed
 bogus-priv
 dhcp-range=${AP_IP_BEGIN}.50,${AP_IP_BEGIN}.150,12h
-
+dhcp-option=3,${AP_IP}
 EOF
-fi
 
-if test true != "${STA_ONLY}"; then
     # Populate `/etc/hostapd/hostapd.conf`
     _logger "Populate /etc/hostapd/hostapd.conf"
-    bash -c 'cat > /etc/hostapd/hostapd.conf' <<EOF
+    bash -c 'test -f /etc/hostapd/hostapd.conf || cat > /etc/hostapd/hostapd.conf' << EOF
 ctrl_interface=/var/run/hostapd
 ctrl_interface_group=0
-interface=ap0
+interface=ap0@wlan0
```

Comment:
should this be `ap@wlan0`?

Reply:
It could be, but convention is to name it apX